fs/blktests.py

```diff
@@ -59,7 +61,14 @@ def test(self):
         os.chdir(self.sourcedir)
 
         genio.write_one_line("/proc/sys/kernel/hung_task_timeout_secs", "0")
-        process.system('./check', ignore_status=True)
+        if self.disk:
+            os.environ['TEST_DEVS'] = self.disk
+            cmd = './check %s' % self.dev_type
+            result = process.run(cmd, ignore_status=True, verbose=True)
+            if result.exit_status != 0:
+                self.fail("test failed")
+        else:
+            process.system('./check', ignore_status=True)
```

Comment:
```
        if self.disk:
            os.environ['TEST_DEVS'] = self.disk
        cmd = './check %s' % self.dev_type
        result = process.run(cmd, ignore_status=True, verbose=True)
        if result.exit_status != 0:
            self.fail("test failed")
```

Reply:
@narasimhan-v actually  test running before like this  './check'. so i left that in else part. if someone don't want to run using disk and type like before they can run.